.github/workflows/ci.yml

```diff
@@ -95,6 +95,11 @@ jobs:
 
       - name: Typecheck with pyright
         run: make pyright
+        if: matrix.python-version != '3.6'
+
+      - name: Typecheck with pyright
```

Comment:
Just skip it?

Reply:
It's fairly useful to run this to make sure that we're not importing things from `typing` that need to be imported from `typing_extensions` in python 3.6. In most cases the test suite would crash, but this would be necessary for `if TYPE_CHECKING`

Reply:
Mind adding a comment about why we have that condition?